Automattic/jetpack
# Prevent adding adding AMP hooks for Infinite Scroll on legacy AMP post templates

Fixes a regression introduced in #16048. See https://github.com/Automattic/jetpack/pull/16048#issuecomment-693008885:

> This appears to not be working properly. AMP validation errors are being emitted on this AMP page, for example: https://alankent.me/2020/06/12/musings-about-5g-and-ar/amp/
> 
> The problem seems to be twofold:
> 
> 1. The `<amp-next-page>` element is erroneously getting added when in legacy Reader mode.
> 2. The `amp-next-page` script is not getting injected because output buffer is being used to inject the tag and/or it's not getting detected during validation because WordPress.com is still on 1.4.x and in that old version the page is not sent through the post-processor. See https://github.com/ampproject/amp-wp/pull/3781.

The validation errors currently occurring on AMP pages on WordPress.com using the Reader mode template when infinite scroll is enabled:

![The tag 'amp-next-page' requires including the 'amp-next-page' extension JavaScript.
line 1260, column 0  
The mandatory reference point 'AMP-NEXT-PAGE > SCRIPT[type=application/json]' for 'amp-next-page with inline config' is missing.
line 1260, column 0  
The tag 'script' requires including the 'amp-next-page' extension JavaScript.
line 1261, column 1
The tag 'template' requires including the 'amp-mustache' extension JavaScript.
line 1268, column 2](https://user-images.githubusercontent.com/134745/93271776-ea6ea280-f768-11ea-8365-9ec548e353a0.png)


#### Changes proposed in this Pull Request:

* Prevent initializing AMP infinite scroll on legacy Reader post templates.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






1. Active AMP plugin
2. Set the template mode to "Reader"
3. Select the AMP Legacy reader theme
4. Enable Infinite Scroll in Jetpack,
5. View an AMP page.
6. See validation errors (on AMP plugin v1.4.x) or else unexpected `amp-next-page` (on AMP plugin v1.5+).

#### Proposed changelog entry for your changes:

* Fix infinite scroll fro erroneously being added on AMP Reader mode post templates.
